### PR TITLE
Feature/network requests async js

### PR DESCRIPTION
**Pull request by RMartin0717**
# Pull Request Template
## Description

Pull data from API instead of dummy data.

## Type of change
Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)

## Checklist:
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [n/a] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] Any dependent changes have been merged and published in downstream modules
- [x] I have checked my code and corrected any misspellings

### CODE REVIEW

File: src/components/App/App.js [hunk 1]
```diff
@@ -5,42 +5,25 @@ import './App.css'
...
+      })
   }
 
   returnHome = (event) => {
-    event.preventDefault();
-    this.setState({ currentMovieID: null })
     this.setState({ currentMovie: null })
   }
 
-  componentDidUpdate() {
```
Comment (errabun) [inline src/components/App/App.js]: Happy we found a solution and got out of the infinite loop we created in this method!

**Review by errabun — COMMENTED** [review]
Great work, love what we have been able to accomplish so far!

File: src/components/App/App.js [hunk 2]
```diff
@@ -2,30 +2,35 @@ import React, { Component } from 'react'
...
 
 class App extends Component {
   constructor() {
-    super()
+    super();
     this.state = {
-      movies: movieData.movies,
-      currentCard: null
+      movies: [],
+      currentMovie: null
```
Comment (sertmer) [inline src/components/App/App.js:11]: Nice! currentMovie is definitely more descriptive.

File: src/components/App/App.js [hunk 3]
```diff
@@ -2,30 +2,35 @@ import React, { Component } from 'react'
...
     }
   }
 
   handleClick = (event) => {
-    event.preventDefault();
-    this.setState({ currentCard: Number(event.target.id) })
+    fetch(`https://rancid-tomatillos.herokuapp.com/api/v2/movies/${Number(event.target.id)}`)
```
Comment (sertmer) [inline src/components/App/App.js:16]: Good stuff! Just don't forget that API calls should be moved into their own file, per the rubric.

File: src/components/App/App.js [hunk 4]
```diff
@@ -2,30 +2,35 @@ import React, { Component } from 'react'
...
-    event.preventDefault();
-    this.setState({ currentCard: null })
+    this.setState({ currentMovie: null })
   }
 
-  findMovie = () => {
-    const currentMovie = this.state.movies.find(movie => movie.id === this.state.currentCard)
-    return currentMovie
+  componentDidMount() {
+    fetch("https://rancid-tomatillos.herokuapp.com/api/v2/movies")
```
Comment (sertmer) [inline src/components/App/App.js:28]: Same here. Be sure to move this fetch into it's own file. Additionally - These two fetches look pretty similar - can you think of any ways to DRY this up with args and params?

**Review by sertmer — COMMENTED** [review]
I left a few notes as comments, but all in all I think you're in a good spot!

**Pull request merged**